docs/developer/designs/node-sets/README.md

```diff
@@ -0,0 +1,159 @@
+# NodeSets
+
+## Overview
+This document outlines an enhancement to the scheduling procedure by introducing the concept of NodeSets, enabling new granularity of pod scheduling over cluster of nodes, by hooking a cluster nodes division logic, and performing scheduling attempts over the resulting sets.
+
+Topology aware scheduling is a good example of such requirement - Attempting to schedule a job over a set of nodes under same topological domain. While such feature can be implemented in pre-predicate and predicate hooks, multi domain is not feasible, limiting the topology scheduling capabilities.
+
+In this document we introduce a new concept of `NodeSet` which is a set of nodes the scheduler attempt to schedule a job over. 
+
+## Motivation
+Current implementation of topology scheduling is limited to a single domain attempt. This mean that when a workload required to be scheduled with topology constraint the plugin searches for a capable domain (Without certainty the job is schedulable over) and try to allocate the job over. If the attempt fails, it will not look for a different domain capable to schedule this workload. 
+
+The uncertainty comes from the fact that the topology plugin can only have a best effort decision, as it does not aware of all scheduling constraints, such as NodeAffinity, LabelSelecot, PVC, etc...
+
+This made us re-think about the integration of the plugin with the scheduling framework, which lead us to suggest the NodeSets concept.
```

Comment:
We knew that we are going to need to tackle it at some point. I am not sure this sentence is necessary.